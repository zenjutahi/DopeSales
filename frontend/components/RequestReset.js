import React, { Component } from "react";
import { Mutation } from "react-apollo";
import FlashMessage from "react-flash-message";
import gql from "graphql-tag";
import Form from "./styles/Form";
import Error from "./ErrorMessage";
import SMessage from "./styles/SMessage";

const REQUEST_RESET_MUTATION = gql`
  mutation REQUEST_RESET_MUTATION($email: String!) {
    requestReset(email: $email) {
      message
    }
  }
`;

class RequestReset extends Component {
  state = {
    email: ""
  };
  saveToState = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  render() {
    return (
      <Mutation mutation={REQUEST_RESET_MUTATION} variables={this.state}>
        {(requestReset, { error, loading, called }) => {
          return (
            <Form
              method="post"
              data-test="form"
              onSubmit={async e => {
                e.preventDefault();
                console.log("******", e.target.value);
                const res = await requestReset();
                this.setState({ email: "" });
              }}
            >
              <fieldset disabled={loading} aria-busy={loading}>
                <h2>Request a password reset</h2>
                <Error error={error} />
                {!error && !loading && called && (
                  <FlashMessage duration={4000}>
                    <SMessage data-test="message">
                      Success! Check your email for the reset link!
                    </SMessage>
                  </FlashMessage>
                )}
                <label htmlFor="email">
                  Email
                  <input
                    type="email"
                    name="email"
                    placeholder="email"
                    value={this.state.email}
                    onChange={this.saveToState}
                  />
                </label>
                <button type="submit">Request Reset</button>
              </fieldset>
            </Form>
          );
        }}
      </Mutation>
    );
  }
}

export default RequestReset;
export { REQUEST_RESET_MUTATION };
