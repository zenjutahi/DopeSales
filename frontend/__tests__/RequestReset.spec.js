import { mount } from "enzyme";
import wait from "waait";
import toJSON from "enzyme-to-json";
import RequestReset, {
  REQUEST_RESET_MUTATION
} from "../components/RequestReset";
import { MockedProvider } from "react-apollo/test-utils";

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: "wesbos@gmail.com" }
    },
    result: {
      data: { requestReset: { message: "success", __typename: "Message" } }
    }
  }
];

describe("<RequestReset/>", () => {
  it("renders and matches snapshot", async () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    const form = wrapper.find('form[data-test="form"]');
    expect(toJSON(form)).toMatchSnapshot();
  });

  it("calls the mutation", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>
    );
    // simulate typing an email
    wrapper.find("input").simulate("change", {
      target: { name: "email", value: "wesbos@gmail.com" }
    });
    // simulate submit the form
    wrapper.find("form").simulate("submit");
    await wait();
    wrapper.update();
    const message = wrapper.find('div[data-test="message"]');
    expect(message.text()).toContain(
      "Success! Check your email for the reset link!"
    );
    expect(toJSON(message)).toMatchSnapshot();
  });
});
