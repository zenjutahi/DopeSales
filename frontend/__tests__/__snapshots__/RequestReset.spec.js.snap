// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> calls the mutation 1`] = `
<div
  className="SMessage-sc-1eh2f1c-0 Kdxbb"
  data-test="message"
>
  Success! Check your email for the reset link!
</div>
`;

exports[`<RequestReset/> renders and matches snapshot 1`] = `
<form
  className="Form-sc-1xszr8q-0 cAFrWY"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Request a password reset
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Request Reset
    </button>
  </fieldset>
</form>
`;
